packages/codecs-data-structures/README.md

```diff
@@ -221,19 +215,57 @@ u32DirectionCodec.encode(Direction.Left); // 0x00000000
 u32DirectionCodec.encode(Direction.Right); // 0x01000000
 ```
 
-Note that if you provide a string enum — e.g. `enum Direction { Left = 'LEFT' }` — to the `getEnumCodec` function, it will only store the index of the variant. However, the string value may be used to encode that index.
+This function also works with lexical enums — e.g. `enum Direction { Left = '←' }` — explicit numerical enums — e.g. `enum Speed { Left = 50 }` — and hybrid enums with a mix of both.
 
 ```ts
-enum Direction {
-    Left = 'LEFT',
-    Right = 'RIGHT',
-    Up = 'UP',
-    Down = 'DOWN',
+enum Numbers {
+    One,
+    Five = 5,
+    Six,
+    Nine = 'nine',
 }
 
-getEnumCodec(Direction).encode(Direction.Right); // 0x01
-getEnumCodec(Direction).encode('Right' as Direction); // 0x01
-getEnumCodec(Direction).encode('RIGHT'); // 0x01
+getEnumCodec(Numbers).encode(Direction.One); // 0x00
+getEnumCodec(Numbers).encode(Direction.Five); // 0x01
+getEnumCodec(Numbers).encode(Direction.Six); // 0x02
+getEnumCodec(Numbers).encode(Direction.Nine); // 0x03
+getEnumCodec(Numbers).encode('One'); // 0x00
+getEnumCodec(Numbers).encode('Five'); // 0x01
+getEnumCodec(Numbers).encode('Six'); // 0x02
+getEnumCodec(Numbers).encode('Nine'); // 0x03
+```
+
+Notice how, by default, the index of the enum variant is used to encode the value of the enum. For instance, in the example above, `Numbers.Five` is encoded as `0x01` even though its value is `5`. This is also true for lexical enums.
+
+However, when dealing with numerical enums that have explicit values, you may use the `useValuesAsDiscriminators` option to encode the value of the enum variant instead of its index.
+
+```ts
+enum Numbers {
+    One,
+    Five = 5,
+    Six,
+    Nine = 9,
+}
+
+const codec = getEnumCodec(Numbers, { useValuesAsDiscriminators: true });
```

Comment:
I had the same thought but that was going to disable hybrid enums which is a shame. One thing I could probably do later on is add more function overloads so that TypeScript would fail if you're trying to use this `useValuesAsDiscriminators` option with an enum that contains lexical values.